src/org/labkey/targetedms/folderImport/QCFolderImporter.java

```diff
@@ -38,55 +42,82 @@ public String getDescription()
     }
 
     @Override
-    public void process(@Nullable PipelineJob job, FolderImportContext ctx, VirtualFile root) throws Exception
+    public boolean isValidForImportArchive(FolderImportContext ctx) throws ImportException
     {
-        if (null != job)
+        //valid only if its a Targeted MS QC folder
+        boolean isTargetedMSFolder = false;
+        boolean isQCFolder = false;
+        if (null != ctx.getXml().getFolderType() && StringUtils.isNotBlank(ctx.getXml().getFolderType().getName()))
+        {
+            if (ctx.getXml().getFolderType().getName().equalsIgnoreCase("Targeted MS"))
+                isTargetedMSFolder = true;
+        }
+        if (isTargetedMSFolder)
         {
-            job.setStatus("IMPORT " + getDescription());
+            ModulePropertyType[] moduleProps = ctx.getDocument().getFolder().getModuleProperties().getModulePropertyArray();
+            List<ModulePropertyType> targetedMSModuleProps = Arrays.stream(moduleProps).filter(modulePropertyType -> modulePropertyType.getPropertyName().equals("TargetedMS Folder Type")).collect(Collectors.toList());
+            if (targetedMSModuleProps.size() == 1 && targetedMSModuleProps.get(0).getValue().equals("QC"))
+            {
+                isQCFolder = true;
+            }
         }
 
-        VirtualFile panoramaQCDir = root.getDir(QCFolderConstants.QC_FOLDER_DIR);
-        List<String> filesToImport = root.getDir(QCFolderConstants.QC_FOLDER_DIR).list();
-        TargetedMSSchema schema = new TargetedMSSchema(ctx.getUser(), ctx.getContainer());
+        return isQCFolder;
+    }
 
-        //iterate through PanoramaQCSettings enum values so that files get imported in that order/ordinal, since the lookup tables need to get populated first
-        for (PanoramaQCSettings qcSetting : PanoramaQCSettings.values())
+    @Override
+    public void process(@Nullable PipelineJob job, FolderImportContext ctx, VirtualFile root) throws Exception
+    {
+        if (isValidForImportArchive(ctx))
```

Comment:
This feels like the wrong approach to me - it will still create an empty directory if there wasn't any QC data in the folder archive.